Homework 3

YuliyaK1 opened the pull request.

Commit:
Вариант до отдельного компонента с элементом списка

Вариант до вынесения элемента списка в отдельный компонент

Commit:
Элемент списка в отдельном компоненте

Элемент списка вынесен в отдельный компонент, настроены input и output между родительским и дочерним компонентами.
Стили из компонента вынесены в общий файл стилей

Commit:
Правки по замечаниям

Правки по замечаниям

Commit:
Правки после исправлений

Правки после исправлений

Commit:
Часть ТЗ про спинер и кнопку

Добавлен спинер, создан компонент кнопки

Commit:
Правки по кнопкам

Внесены изменения по компоненту с кнопкой и стилям

Commit:
eslint

Добавление eslint и правила "comma-dangle": [2, "always-multiline"].
А также исправление замечаний по comma-dangle.

Commit:
eslint

eslint

gearmobile commented on src/app/to-do-list/to-do-list.component.html:
```diff
@@ -3,24 +3,18 @@ <h1 class="header">Список дел</h1>
-                <li>Добавить разметку в шаблоны</li>
-                <button class="button-del">Удалить</button>                
+            <ul>                
+                <li *ngFor="let listItem of items">      
```
замените `*ngFor`  на `@if`

YuliyaK1 replied:
Исправлено

gearmobile commented on src/app/to-do-list/to-do-list.component.html:
```diff
@@ -3,24 +3,18 @@ <h1 class="header">Список дел</h1>
-            <input type="text" placeholder="Добавить новое задание">  
-            <button class="button-add">Добавить</button>
+            <input matInput type="text" placeholder="Добавить новое задание" [(ngModel)]="newItem" (ngModelChange)="editInput(newItem)">  
+            <button [ngClass]="{'disabled': !ActiveAdd}" class="button-add" (click)="addItem(newItem)">Добавить</button>
```
замените ngClass - на style direct binding:

```
[class.disabled]="!ActiveAdd"
```

... так как ngClass - уже decrecated.

gearmobile commented on src/app/to-do-list/to-do-list.component.ts:
```diff
@@ -1,11 +1,57 @@
+  {id: 3, name: 'Сложить числа'},
+]
+
+ActiveAdd: boolean = false;
+newItem: string = '';
+idMax: number = 0;
```
бесполезные аннотации типа; лучше так, например:

```
ActiveAdd = false
```

gearmobile commented on src/app/to-do-list-item/to-do-list-item.component.html:
```diff
@@ -0,0 +1,3 @@
+<li *ngIf="itemsList"> 
```
замените *ngIf - на @if

gearmobile commented on src/app/to-do-list-item/to-do-list-item.component.ts:
```diff
@@ -9,4 +10,15 @@ import { CommonModule } from '@angular/common';
 export class ToDoListItemComponent {
 
+  @Input() itemsList: TItems| undefined;
+  @Output() dltItem = new EventEmitter<number>();
```
замените на сигнальные input и output

gearmobile commented on src/app/to-do-list-item/to-do-list-item.component.ts:
```diff
@@ -9,4 +10,15 @@ import { CommonModule } from '@angular/common';
+    if (this.itemsList)
+    {
+      this.dltItem.emit(id);
+      console.log(this.itemsList);
```
вообще-то принято удалять все `console` - перед отправкой на код-ревью

gearmobile commented on src/app/to-do-list/to-do-list.component.html:
```diff
@@ -3,11 +3,11 @@ <h1 class="header">Список дел</h1>
-                </li>         
+            <ul>
+                    <app-to-do-list-item 
+                        *ngFor="let listItem of items; let i = index"
```
замените `*ngFor` - на `@for`

gearmobile commented on src/app/button/button.component.ts:
```diff
@@ -0,0 +1,13 @@
+})
+export class ButtonComponent {
+
+  @Input() title: string='123';
```
сигнальный input - вместо `@Input`

gearmobile commented on src/app/to-do-list-item/to-do-list-item.component.html:
```diff
@@ -1,3 +1,3 @@
-<li *ngIf="itemsList"> 
-    {{itemsList.name}}<button class="button-del" (click)="deleteItem(itemsList.id)">Удалить</button> 
+<li *ngIf="itemsList">  
```
`@if` вместо `*ngIf`

gearmobile commented on src/app/to-do-list/to-do-list.component.html:
```diff
@@ -4,16 +4,21 @@ <h1 class="header">Список дел</h1>
             <ul>
+                
+                <div class="spinner" *ngIf="isLoading"></div>
+                <div *ngIf="!isLoading">              
```
`@if @else` - такая конструкция здесь подойдет лучше и современней

gearmobile commented on src/app/to-do-list/to-do-list.component.html:
```diff
@@ -4,16 +4,21 @@ <h1 class="header">Список дел</h1>
     </div>
         <div class="container fl">
             <input matInput type="text" placeholder="Добавить новое задание" [(ngModel)]="newItem" (ngModelChange)="editInput(newItem)">  
+            <app-button></app-button>
```
в подобный случаях принято писать сокращенный вариант - `<app-button />`

gearmobile commented on src/app/to-do-list/to-do-list.component.ts:
```diff
@@ -22,6 +23,7 @@ items: TItems[] = [
 ActiveAdd: boolean = false;
 newItem: string = '';
 idMax: number = 0;
+isLoading: boolean = true;
```
бесполезная аннотация

gearmobile commented on src/app/button/button.component.ts:
```diff
@@ -8,6 +8,6 @@ import { Component, Input } from '@angular/core';
 export class ButtonComponent {
 
-  @Input() title: string='123';
+  @Input() titleButton: string='';
```
сигнальный input

gearmobile commented on src/app/to-do-list/to-do-list.component.html:
```diff
@@ -18,8 +18,7 @@ <h1 class="header">Список дел</h1>
             <input matInput type="text" placeholder="Добавить новое задание" [(ngModel)]="newItem" (ngModelChange)="editInput(newItem)">  
-            <app-button></app-button>
-            <button [ngClass]="{'disabled': !ActiveAdd}" class="button-add" (click)="addItem(newItem)">Добавить</button>
+            <app-button [ngClass]="{'disabled': !ActiveAdd}" class="button-add" (click)="addItem(newItem)" [titleButton]="'Добавить'"></app-button>
```
style direct binding, `ngClass` - устарел

Commit:
Исправления по замечания по ДЗ 3

Замена:
*ngIf на @if @else, *ngFor на @for
@Input, @Output на сигнальные
 ngClass на style direct binding

YuliyaK1 commented:
Исправлены все замечания кроме замечания про "бесполезные аннотации типа". 
Можно использовать динамическое определение типов, но использование явных типов делает код более понятным и читабельным, к тому же уменьшает вероятность переопределения переменной на некорректное значение, из-за которого могут появится нестабильные и трудноуловимые ошибки.
Поэтому не могу согласиться с данным замечанием.
По замечания про console, эти строки закомментировала без удаления, бывают нужны для отладки
